core/src/test/java/org/apache/iceberg/jdbc/TestJdbcUtil.java

```diff
@@ -18,14 +18,116 @@
+          .prepareStatement(
+              "CREATE TABLE "
+                  + JdbcUtil.CATALOG_TABLE_VIEW_NAME
+                  + "("
+                  + JdbcUtil.CATALOG_NAME
+                  + " VARCHAR(255) NOT NULL,"
+                  + JdbcUtil.TABLE_NAMESPACE
+                  + " VARCHAR(255) NOT NULL,"
+                  + JdbcUtil.TABLE_NAME
+                  + " VARCHAR(255) NOT NULL,"
+                  + JdbcTableOperations.METADATA_LOCATION_PROP
+                  + " VARCHAR(1000),"
+                  + JdbcTableOperations.PREVIOUS_METADATA_LOCATION_PROP
+                  + " VARCHAR(1000),"
+                  + "PRIMARY KEY("
+                  + JdbcUtil.CATALOG_NAME
+                  + ","
+                  + JdbcUtil.TABLE_NAMESPACE
+                  + ","
+                  + JdbcUtil.TABLE_NAME
+                  + "))")
+          .executeUpdate();
+      connection
+          .prepareStatement(
+              "INSERT INTO "
+                  + JdbcUtil.CATALOG_TABLE_VIEW_NAME
+                  + "("
+                  + JdbcUtil.CATALOG_NAME
+                  + ","
+                  + JdbcUtil.TABLE_NAMESPACE
+                  + ","
+                  + JdbcUtil.TABLE_NAME
+                  + ","
+                  + JdbcTableOperations.METADATA_LOCATION_PROP
+                  + ","
+                  + JdbcTableOperations.PREVIOUS_METADATA_LOCATION_PROP
+                  + ") VALUES('TEST','namespace1','table1',null,null)")
+          .execute();
+      connection
+          .prepareStatement(
+              "INSERT INTO "
+                  + JdbcUtil.CATALOG_TABLE_VIEW_NAME
+                  + "("
+                  + JdbcUtil.CATALOG_NAME
+                  + ","
+                  + JdbcUtil.TABLE_NAMESPACE
+                  + ","
+                  + JdbcUtil.TABLE_NAME
+                  + ","
+                  + JdbcTableOperations.METADATA_LOCATION_PROP
+                  + ","
+                  + JdbcTableOperations.PREVIOUS_METADATA_LOCATION_PROP
+                  + ") VALUES('TEST','namespace2','table2',null,null)")
+          .execute();
+
+      // update the schema
+      connection.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
+
+      // we check if the type column has been added
+      DatabaseMetaData dbMeta = connection.getMetaData();
+      ResultSet columns =
+          dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.TYPE);
+      Assertions.assertThat(columns.next()).isTrue();
+
+      // we check the JDBC catalog supports schema from previous version to list tables
+      ResultSet tablesRS = connection.prepareStatement(JdbcUtil.LIST_TABLE_SQL).executeQuery();
+      Map<String, String> tables = Maps.newHashMap();
+      while (tablesRS.next()) {
+        tables.put(
+            tablesRS.getString(JdbcUtil.TABLE_NAME), tablesRS.getString(JdbcUtil.TABLE_NAMESPACE));
+      }
+
+      Assertions.assertThat(tables.size()).isEqualTo(2);
```

Comment:
```suggestion
      assertThat(tables).hasSize(2);
```

Reply:
in fact you can combine all three assertions into one: `assertThat(tables).hassize(2).containsEntry("namespace1", "table1").containsEntry("namespace2", "table2")`